sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -60,7 +60,15 @@ public UcfgInstructionFactory(SemanticModel semanticModel, UcfgExpressionService
         public IEnumerable<Instruction> CreateFromAttributeSyntax(AttributeSyntax attributeSyntax, IMethodSymbol attributeCtor,
             string parameterName)
         {
-            var targetOfAttribute = expressionService.GetExpression(attributeSyntax.Parent.Parent);
+            // Only variable expressions can be annotated.
+            // Constants (e.g. parameters of type int) cannot be annotated.
+            var targetOfAttribute = expressionService.GetExpression(attributeSyntax.Parent.Parent)
```

Comment:
It seemed cleaner to do it here since the logic is related to handling attributes.
_CreateFunctionCall_ is already handling quite a lot. It shouldn't need to know anything about attributes. However, perhaps we could add some more general sanity checks e.g. that expressions aren't null.